main.c

```diff
@@ -1391,6 +1391,12 @@ main
   if (repeat_error && ErrorBufMessage)
     puts(ErrorBuf);
 main_exit:
+  if (NeoMutt && NeoMutt->sub)
```

Comment:
1. They weren't removed before.
   They were simply cleaned up when the NeoMutt object was deleted.
2. In `main()`s cleanup, yes we could, but if possible I'd rather be explicit:
   `alloc_x();` ... `free_x();`

Reply:
> 1. They weren't removed before.

ok, that's what I thought too

> 2. In main()s cleanup, yes we could, but if possible I'd rather be explicit:
alloc_x(); ... free_x();

Valid point. My counter argument would be that `notify_observer_remove_all()` leads to one place less I need to remember to change. But I'm nitpicking now since we're talking about the "optional" `main()` cleanup.